lib/util/stream_utils.js

```diff
@@ -567,11 +608,60 @@ shaka.util.StreamUtils = class {
       channelsCount: null,
       audioSamplingRate: null,
       spatialAudio: false,
+      tilesLayout: null,
       audioBandwidth: null,
       videoBandwidth: null,
       originalVideoId: null,
       originalAudioId: null,
       originalTextId: stream.originalId,
+      originalImageId: null,
+    };
+
+    return track;
+  }
+
+
+  /**
+   * @param {shaka.extern.Stream} stream
+   * @return {shaka.extern.Track}
+   */
+  static imageStreamToTrack(stream) {
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+
+    /** @type {shaka.extern.Track} */
+    const track = {
+      id: stream.id,
+      active: false,
+      type: ContentType.IMAGE,
+      bandwidth: stream.bandwidth || 0,
+      language: '',
+      label: null,
+      kind: null,
+      width: stream.width || null,
+      height: stream.height || null,
+      frameRate: null,
+      pixelAspectRatio: null,
+      hdr: null,
+      mimeType: stream.mimeType,
+      codecs: null,
+      audioCodec: null,
+      videoCodec: null,
+      primary: false,
+      roles: [],
+      audioRoles: null,
+      forced: false,
+      videoId: null,
+      audioId: null,
+      channelsCount: null,
+      audioSamplingRate: null,
+      spatialAudio: false,
+      tilesLayout: stream.tilesLayout || null,
```

Comment:
Is it possible for an image stream not to have this?

Reply:
In the case of HLS, as it is possible that it works differently, it is possible that the information does not come at the playlist level, but at the segment level. See: https://github.com/google/shaka-player/pull/3145#discussion_r579072813